draft-ietf-quic-recovery.md

```diff
@@ -824,37 +824,70 @@ packets.
-experiencing persistent congestion.  Commonly, this can be established by
-consecutive PTOs, but since the PTO timer is reset when a new ack-eliciting
-packet is sent, an explicit duration must be used to account for those cases
-where PTOs do not occur or are substantially delayed. The rationale for this
-threshold is to enable a sender to use initial PTOs for aggressive probing, as
-TCP does with Tail Loss Probe (TLP; see {{RACK}}), before establishing
-persistent congestion, as TCP does with a Retransmission Timeout (RTO; see
-{{?RFC5681}}). The RECOMMENDED value for kPersistentCongestionThreshold is 3,
-which is approximately equivalent to two TLPs before an RTO in TCP.
-
-The persistent congestion period SHOULD NOT start until there is at
-least one RTT sample.  Prior to an RTT sample, the duration cannot be
-correctly calculated.  Waiting for one RTT sample also avoids spuriously
-declaring persistent congestion when the initial RTT is larger than the
-actual RTT.
-
-This duration is computed as follows:
+When a sender establishes loss of all in-flight packets sent over a long enough
+duration, the network is considered to be experiencing persistent congestion.
+
+### Duration {#pc-duration}
+
+The persistent congestion duration is computed as follows:
 
 ~~~
 (smoothed_rtt + max(4*rttvar, kGranularity) + max_ack_delay) *
     kPersistentCongestionThreshold
 ~~~
 
-Unlike the PTO computation in {{pto}}, persistent congestion includes the
-max_ack_delay irrespective of the packet number spaces in which losses are
-established.
+Unlike the PTO computation in {{pto}}, this duration includes the max_ack_delay
+irrespective of the packet number spaces in which losses are established.
+
+This duration allows a sender to send as many packets, including some in
+response to PTO expiration, as TCP does with Tail Loss Probe ({{RACK}}), before
+establishing persistent congestion, as TCP does with a Retransmission Timeout
+({{?RFC5681}}).
+
+The RECOMMENDED value for kPersistentCongestionThreshold is 3, which is
+approximately equivalent to two TLPs before an RTO in TCP.
+
+This design does not use consecutive PTO events to establish persistent
+congestion, since application patterns impact PTO expirations. For example, a
+sender that sends small amounts of data with silence periods between them
+restarts the PTO timer every time it sends, potentially preventing the PTO timer
+from expiring for a long period of time, even when no acknowledgments are being
+received. The use of a duration enables a sender to establish persistent
+congestion without depending on the occurrence of PTOs.
+
+### Establishing Persistent Congestion
+
+A sender establishes persistent congestion on receiving an acknowledgement if at
+least two ack-eliciting packets are declared lost, and:
+
+* a prior RTT sample existed when both packets were sent;
+
+* the duration between the send times of these two packets exceeds the
```

Comment:
I think this is still ambiguous if more than 2 packets are lost, so I'm reusing language below.

```suggestion
* the duration between the send times of the oldest and newest lost packet exceeds the
```

Reply:
That's still not adequate. The point here is that there's a contiguous period that is long enough, and the duration has to be between two packets that were marked lost, not necessarily the oldest and the newest packets. For example, consider that 
- packets 1-10 were sent,
- all odd packets were marked lost: 1, 3, 5, 7, 9.
- the duration between 1 and 9 was greater than the persistent congestion duration.

According to the existing design, this should not be considered persistent congestion, but with your proposed text, it will be.

Reply:
There's a requirement that all packets between the largest and smallest are marked lost as well, so my suggested change would still not declare this as persistent congestion(assuming 2, 4, 6 or 8 were acknowledged).

I still think this is unclear if more than 2 packets are lost, so maybe you have a suggestion?

Reply:
We can't require all packets between largest and smallest be acked; see my comment: https://github.com/quicwg/base-drafts/pull/3961#discussion_r463907388.

Reply:
Sorry for the confusion, I now have what you have in mind.

If you put this bullet after the next one, I think this will be clearer.

Reply:
Actually, I might move that third bullet to be first.  Conceptually, it frames the idea that they're the first and last packets of a 2+ packet sequence.

Reply:
Yup, suggestions taken.